src/data/sonata_data.cpp

```diff
@@ -28,7 +28,9 @@ SonataData::SonataData(const std::string& report_name,
     , nodes_(nodes) {
     prepare_buffer(max_buffer_size);
     index_pointers_.resize(nodes->size());
-    time_ = {tstart, tend, dt};
+
+    // Round the tstart value to avoid conflicts in case of save-restore
+    time_ = {round(tstart / dt) * dt, tend, dt};
```

Comment:
At high level, reporting library shouldn't worry about this aspect and shouldn't change time like this. Instead, simulator should provide the correct/expected value? (and then we avoid such change in multiple libraries i.e. including reportinglib)

Reply:
Yes, but if we do this, the whole simulation is different and bitwise inconsistent compared to running it from scratch without save-restore. In other words, if the `tstart` is rounded for the simulation, it will be different in the restore.

Reply:
I find this a bit concerning - which I think mirrors @pramodk's point.

Also, going back to the original CNEUR-335, I find it interesting that the 'fix' here involves changing the start time, but we're seeing strange results at the end time - I don't see how the 'adjusted' start time will match the end time that triggered the original ticket.

Finally, is the `rounding` behavior correct in all cases? On even, round moves away from zero, and I'm wondering if this will cause a gap or an overlap, depending on how the simulator's internal time is saved.

Reply:
> What do you mean that you are seeing strange results at the end time?

Ah, right, forgot the exact problem.

> If you mean the end time of the first simulation (save) this is stored in the header after reading the BlueConfig

What happens if someone chooses a split time of `3999.9`?  Doesn't that mean the first sim save will have `3999.9`, but then the start time of the next sim will be `4000`?

Reply:
> What happens if someone chooses a split time of 3999.9? Doesn't that mean the first sim save will have 3999.9, but then the start time of the next sim will be 4000?
No, we round to the closest dt, in that case the tend of the initial sim will be 3999.9 that will be read by second sim and will set tstart = round(3.999/0.1)*0.1 = 3999.9

Reply:
Ok, with a `dt=0.1`, what if the value is `3999.9999`?

Or, in my original example, the `dt=0.025` and the split time is `3999.9`?

Reply:
The minimum possible reporting dt is 0.025, that is the minimum simulation dt.
This means that there will be a new timestep recorded every 0.025 ms. Since the report records [tstart, tend) in your example, the last timesteps recorded would be:
3999.925
3999.950
3999.975
So basically 3999.999 doesnt make sense as it is, the closer rounded to the dt would be 4000

And if your concern is inconsistencies between tend of one simulation and tstart of another, we round to the closest dt both tstart and tend and there wont be any inconsistencies anymore.

Reply:
> So basically 3999.999 doesnt make sense as it is

Will it be rejected when parsing the blueconfig? I'm reading the above as the value stored comes directly from the blueconfig so an inconsistency would be created, no?

Reply:
Any feed back on `Will it be rejected when parsing the blueconfig`?